gridstatus/ieso.py

```diff
@@ -0,0 +1,517 @@
+MAXIMUM_DAYS_IN_FUTURE_FOR_LOAD_FORECAST = 1
+
+"""ZONAL LOAD FORECAST CONSTANTS"""
+ZONAL_LOAD_FORECAST_INDEX_URL = "http://reports.ieso.ca/public/OntarioZonalDemand"
+
+# Each forecast file contains data from the day in the filename going forward for
+# 34 days. The most recent file does not have a date in the filename.
+ZONAL_LOAD_FORECAST_TEMPLATE_URL = (
+    f"{ZONAL_LOAD_FORECAST_INDEX_URL}/PUB_OntarioZonalDemand_YYYYMMDD.xml"
+)
+
+# The farthest in the past that forecast files are available
+MAXIMUM_DAYS_IN_PAST_FOR_ZONAL_LOAD_FORECAST = 90
+# The farthest in the future that forecasts are available. Note that there are not
+# files for these future forecasts, they are in the current day's file.
+MAXIMUM_DAYS_IN_FUTURE_FOR_ZONAL_LOAD_FORECAST = 34
+
+
+MINUTES_INTERVAL = 5
+MINUTES_INTERVALS_IN_AN_HOUR = 12
+HOUR_INTERVAL = 1
+
+# Default namespace used in the XML files
+NAMESPACES_FOR_XML = {"": "http://www.ieso.ca/schema"}
+
+
+class IESO(ISOBase):
+    """Independent Electricity System Operator (IESO)"""
+
+    name = "Independent Electricity System Operator"
+    iso_id = "ieso"
+
+    # All data is provided in EST, and does not change with DST. This means there are
+    # no repeated or missing hours in the raw data and we can safely use tz_localize
+    # without setting ambiguous or nonexistent times.
+    # https://www.ieso.ca/-/media/Files/IESO/Document-Library/engage/ca/ca-Introduction-to-the-Capacity-Auction.ashx
+    default_timezone = "EST"
+
+    status_homepage = "https://www.ieso.ca/en/Power-Data"
+
+    def get_load(self, date, end=None, verbose=False):
+        """
+        Get 5-minute load for the Market and Ontario for a given date or from
+        date to end date.
+
+        Args:
+            date (datetime.date | datetime.datetime | str): The date to get the load for
+                Can be a `datetime.date` or `datetime.datetime` object, or a string
+                with the values "today" or "latest". If `end` is None, returns
+                only data for this date.
+            end (datetime.date | datetime.datetime, optional): End date. Defaults None
+                If provided, returns data from `date` to `end` date. The `end` can be a
+                `datetime.date` or `datetime.datetime` object.
+            verbose (bool, optional): Print verbose output. Defaults to False.
+            frequency (str, optional): Frequency of data. Defaults to "5min".
+
+        Returns:
+            pd.DataFrame: zonal load as a wide table with columns for each zone
+        """
+        # We have to add special time handling here because the existing utils
+        # don't perfectly handle the 1 hour data files.
+        if isinstance(date, tuple):
+            date, end = date
+
+        today = utils._handle_date("today", tz=self.default_timezone)
+
+        # Return data from the earliest interval today to the latest interval today
+        if date in ["today", "latest"]:
```

Comment:
the handling for "latest" should be to download this url: `http://reports.ieso.ca/public/RealtimeConstTotals/PUB_RealtimeConstTotals.xml` which always points to most recent interval

Reply:
So only return (at most) 1 hour for "latest"?

Reply:
ya, the idea behind "latest" is to make the fastest possible request that returns the latest data.